Remove Sophgo-specific -mno-ldd flag for standard GCC

The -mno-ldd flag is a Sophgo GCC extension for C906 LDD performance
workaround. Standard GCC 14 does not support this flag.

--- a/arch/riscv/Makefile
+++ b/arch/riscv/Makefile
@@ -30,10 +30,6 @@
 	KBUILD_AFLAGS += -mabi=lp64
 
-# workaround for C906 ldd performance issue
-ifeq ($(CONFIG_ARCH_CVITEK),y)
-	KBUILD_CFLAGS += -mno-ldd
-endif
 
 	KBUILD_LDFLAGS += -melf64lriscv
 else
